lib/pact/command/cont_command.ex

```diff
@@ -0,0 +1,330 @@
+      ) do
+    with {:ok, payload} <- create_payload(cmd_request),
+         {:ok, cmd} <- command_to_json_string(payload, cmd_request),
+         {:ok, sig_commands} <- sign_commands([], cmd, keypairs),
+         {:ok, hash} <- get_unique_hash(sig_commands),
+         {:ok, signatures} <- build_signatures(sig_commands, []),
+         {:ok, command} <- create_command(hash, signatures, cmd) do
+      %Command{} = command
+    end
+  end
+
+  def build(_module), do: {:error, [exec_command_request: :invalid_payload]}
+
+  @spec create_payload(t()) :: valid_payload()
+  defp create_payload(%__MODULE__{
+         data: data,
+         pact_tx_hash: pact_tx_hash,
+         proof: proof,
+         rollback: rollback,
+         step: step
+       }) do
+    [data: data, pact_id: pact_tx_hash, proof: proof, rollback: rollback, step: step]
+    |> ContPayload.new()
+    |> PactPayload.new()
+    |> (&{:ok, &1}).()
+  end
+
+  @spec create_command(
+          hash :: hash(),
+          sigs :: signatures(),
+          cmd :: cmd()
+        ) :: valid_command()
+  defp create_command(hash, sigs, cmd) do
+    case Command.new(hash: hash, sigs: sigs, cmd: cmd) do
+      %Command{} = command -> {:ok, command}
+    end
+  end
+
+  @spec command_to_json_string(payload :: pact_payload(), t()) :: valid_command_json_string()
+  defp command_to_json_string(payload, %__MODULE__{
+         network_id: network_id,
+         meta_data: meta_data,
+         signers: signers,
+         nonce: nonce
+       }) do
+    [
+      network_id: network_id,
+      payload: payload,
+      meta: meta_data,
+      signers: signers,
+      nonce: nonce
+    ]
+    |> CommandPayload.new()
+    |> JSONPayload.parse()
+    |> (&{:ok, &1}).()
+  end
+
+  @spec sign_commands(signs :: list(), cmd :: json_string_payload(), keypairs()) ::
+          valid_sign_commands()
+  defp sign_commands(signs, _cmd, []), do: {:ok, signs}
+
+  defp sign_commands(signs, cmd, [%KeyPair{} = keypair | keypairs]) do
+    signs
+    |> sign_command(cmd, keypair)
+    |> sign_commands(cmd, keypairs)
+  end
+
+  @spec sign_command(signs :: list(), cmd :: json_string_payload(), keypair()) ::
+          list()
+  defp sign_command(signs, cmd, %KeyPair{} = keypair) do
+    {:ok, sign_command} = Sign.sign(cmd, keypair)
+    signs ++ [sign_command]
+  end
+
+  @spec build_signatures(sign_commands :: sign_commands(), result :: list()) :: valid_signatures()
+  defp build_signatures([], result), do: {:ok, SignaturesList.new(result)}
+
+  defp build_signatures([%SignCommand{sig: sig} | rest], result),
+    do: build_signatures(rest, result ++ [sig])
+
+  @spec get_unique_hash(sign_commands()) :: valid_hash()
+  defp get_unique_hash(sign_commands) do
+    unique_hash =
+      Enum.reduce(sign_commands, nil, fn %SignCommand{hash: hash}, acc ->
+        if is_nil(acc) or acc == hash,
+          do: hash,
+          else: {:error, [hash: :hashes_are_not_unique]}
+      end)
+
+    {:ok, unique_hash}
+  end
```

Comment:
@EdwinGuayacan @Odraxs this is duplicate code, which means you have to abstract it from the Command implementations

Reply:
@juanhurtado10  Does it require a separate utils/helper file, or can it be directly abstracted in the `command behavior`?

Reply:
@Odraxs The Utils/Helper modules have a problem ... all the code that doesn't fit in a module ends up in Utils, so my suggestion is to create a specific module to do this